src/drivers/shiftx_drv.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "shiftx_drv.h"
+#include "CAN.h"
+#include "printk.h"
+#include "api_event.h"
+
+#define _LOG_PFX "[ShiftX] "
+
+#define DEFAULT_CAN_TIMEOUT   100
+#define CONFIG_MESSAGE_OFFSET 3
+
+#define ANNOUNCEMENT_OFFSET 0
+
+#define CONFIG_MESSAGE_SET_DISCRETE_LED_OFFSET 10
+
+#define CONFIG_MESSAGE_SET_ALERT_OFFSET 20
+#define CONFIG_MESSAGE_SET_ALERT_THRESHOLD_OFFSET 21
+#define CONFIG_MESSAGE_UPDATE_ALERT_VALUE_OFFSET 22
+
+#define CONFIG_MESSAGE_CONFIGURE_LINEAR_GRAPH_OFFSET 40
+#define CONFIG_MESSAGE_SET_LINEAR_THRESHOLD_OFFSET 41
+#define CONFIG_MESSAGE_UPDATE_LINEAR_GRAPH_VALUE_OFFSET 42
+
+#define CONFIG_MESSAGE_SET_DISPLAY_OFFSET 50
+
+#define NOTIFICATION_BUTTON_STATE_OFFSET 60
+
+static struct shiftx_configuration shiftx_config = {0, 0, 1, 0xE3600, 51};
+
+static struct {
+        bool received;
+        uint8_t id;
+        uint8_t state;
+} button_state = {0};
```

Comment:
The initialization is unnecessary as it is a static struct and thus will be initialized to zero as part of BSS.  Unless you want this in data section for some reason.